rasa/utils/train_utils.py

```diff
@@ -556,9 +556,7 @@ def _check_loss_setting(component_config: Dict[Text, Any]) -> None:
     ):
         rasa.shared.utils.io.raise_warning(
             f"{CONSTRAIN_SIMILARITIES} is set to `False`. It is recommended "
-            f"to set it to `True` when using cross-entropy loss. It will be set to "
-            f"`True` by default, "
-            f"Rasa Open Source 3.0.0 onwards.",
+            f"to set it to `True` when using cross-entropy loss.",
```

Comment:
So this one seems a bit weird to me, since I figured if it's forced to `True` we wouldn't really need to warn at all. I looked into it, and this function is only used it one place (`validate_configuration_settings`), and ultimately is only called from `DIETClassifier` (which has a default value of `False` for `CONSTRAIN_SIMILARITIES`). I see when I try `rasa train` I don't get a warning, and it looks like the default pipeline has it set to `True`, but if we're going to warn when people set it to `False` is there a reason we let them? @samsucik do you have some more context on this?

Reply:
@usc-m @carlad honestly, I'm confused by the original warning. Like, what does it mean `It will be set to True by default, Rasa Open Source 3.0.0 onwards.`? Does that mean that in 3.0.0 we'll override whatever user-set value with `True`, or that the default pipelines will set it to `True` but the user can still override it? @dakshvar22 maybe you know 🙂 If my first interpretation above is correct, then I'd change & rename `_check_loss_setting` such that it sets `constrain_similarities=True` and perhaps informs the user. If the 2nd interpretation is correct, I'd go with the warning as updated by @carlad.

Reply:
> or that the default pipelines will set it to True but the user can still override it?

This